aaad/aaad_app.py

```diff
@@ -0,0 +1,159 @@
+from flask import Flask, jsonify, request, render_template, redirect, make_response, flash, Response
+from flask_restful import abort, Api, Resource
+from flask_login import login_user, logout_user, login_required, current_user
+from webargs import fields
+from webargs.flaskparser import use_args
+import os
+import json
+import logging
+
+from authenticators import FileAuthenticator
+from authorizers import FileAuthorizer
+from resources import Users, Groups, CanActAsUsers
+from sessions import login_manager, SessionUser
+import sessions
+
+app = Flask(__name__)
+app.secret_key = os.environ['APP_SECRET_KEY']
+api = Api(app)
+login_manager.init_app(app)
+
+@app.before_first_request
+def setup_logging():
+    if not app.debug:
+        handler = logging.StreamHandler() # sys.stderr
+        formatter = logging.Formatter('[%(asctime)-15s] %(levelname)s - %(name)s - IP:%(clientip)s User:%(user)s ActAs:%(act_as)s - %(message)s')
+        handler.setFormatter(formatter)
+        app.logger.addHandler(handler)
+        log_levels = {'debug': logging.DEBUG,
+              'info': logging.INFO,
+              'warning': logging.WARNING,
+              'error': logging.ERROR,
+              'critical': logging.CRITICAL}
+        log_level = os.environ.get('LOG_LEVEL', 'warning')
+        app.logger.setLevel(log_levels.get(log_level, logging.NOTSET))
+
+@app.route('/auth')
+@login_required
+def authorize():
+    '''
+    This method finds out if the current user is authorized to -
+        - perform the action specified in 'method' header
+        - on the resource specified in 'URI header'
+    The request body and content_type are also required for ths purpose.
+
+    Note:
+    Because of the @login_required decorator authentication should be already complete
+    using one of the supported methods by the time we reach this method. So, let's just
+    worry about authorizaton.
+    '''
+    user = current_user.get_username()
+    actas = request.cookies.get("actas") # try to get it from cookie
+    if not actas:
+        actas = request.headers.get("act_as_user") # try to get from header
+    if not actas:
+        actas = user # default to the user
```

Comment:
They don't have to be. I wanted to use a smaller name for the cookies. But didn't change the header. 
I'm hoping we'll stop using the basic auth + this header and switch over to cookie/token based auth entirely.